### PR TITLE
Use `orchestra/testbench-core` as `testbench` should only be used for Package development

### PR DESCRIPTION
**Pull request by crynobone**
In package development, `orchestra/testbench` should be added explicitly which then requires `orchestra/workbench`, `orchestra/canvas` etc which isn't useful for this package usage when used in Laravel project.

### CODE REVIEW

**Comment by alies-dev** [comment]
thank you, @crynobone ! I'll take a look into it soon, but firstly would like to understand why orchestra packages are not in dev requirements. It will take some time, but I'll do my best, thanks

**Comment by crynobone** [comment]
> but firstly would like to understand why orchestra packages are not in dev requirements.

It is being used directly via the following class:

https://github.com/psalm/psalm-plugin-laravel/blob/edc45c0cd193c3e82c3503154a56cb581e4b1e20/src/Providers/ApplicationProvider.php#L11-L23

So changing to `orchestra/testbench-core` will be the most minimal changes to avoid loading requirements available via `orchestra/testbench` when using this package.

File: composer.json [hunk 1]
```diff
@@ -24,7 +25,7 @@
         "illuminate/support": "^9.48 || ^10.0",
         "illuminate/view": "^9.48 || ^10.0",
         "nikic/php-parser": "^4.13",
-        "orchestra/testbench": "^7.19 || ^8.0",
+        "orchestra/testbench-core": "^7.35 || ^8.15",
```
Comment (alies-dev) [inline composer.json]: >^7.35 || ^8.15"

Do these versions contain for required functionality, or are they just to enforce new versions? 

Note, this plugin built to support old versions

File: composer.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
         "php": "^8.0.2",
         "ext-simplexml": "*",
         "barryvdh/laravel-ide-helper": "^2.13",
+        "fakerphp/faker": "^1.21",
```
Comment (alies-dev) [inline composer.json]: I couldn't find places where we use Faker on this package. Do you know them?

File: composer.json [hunk 3]
```diff
@@ -34,6 +35,7 @@
         "codeception/module-cli": "^2.0",
         "codeception/module-filesystem": "^3.0",
         "codeception/module-phpbrowser": "^3.0",
+        "laravel/framework": "^9.48 || ^10.0",
```
Comment (alies-dev) [inline composer.json:37]: this package is made to support both Laravel and Lumen. I'm not sure that require Laravel is a good idea.

**Review by alies-dev — COMMENTED** [review]
I have some questions regarding these changes in dependencies, can you please take a look?

File: composer.json (hunk 1 above)
Comment (crynobone) [reply composer.json]: Both versions contain conflict restrictions (using composer.json) to ensure Testbench Core 7 will only be used with Laravel 9 and Testbench Core 8 only used with Laravel 10.

File: composer.json (hunk 3 above)
Comment (crynobone) [reply composer.json:37]: This is only require-dev deps. Previously was provided in `require` via `orchestra/testbench`. 

So this actually improves support for Lumen.

File: composer.json (hunk 2 above)
Comment (crynobone) [reply composer.json]: https://github.com/psalm/psalm-plugin-laravel/blob/edc45c0cd193c3e82c3503154a56cb581e4b1e20/stubs/legacy-factories/FactoryBuilder.stubphp#L5

**Comment by alies-dev** [comment]
Great, thank you, @crynobone 
I see some issues caused updates on the trunk branch, but I'll fix them on my own.

**Pull request merged**